src/charts/time_series_labeler.coffee

```diff
@@ -56,121 +99,143 @@ Ember.Charts.TimeSeriesLabeler = Ember.Mixin.create
-        when 'weeks' , 'W'  then 'week'
-        when 'hours' , 'H' then 'hour'
-        when 'seconds', 'S' then 'second'
+      interval = @domainTypeToLongDomainTypeSingular (@get 'xAxisTimeInterval')
       if interval is 'quarter'
-        count = 3
+        count = MONTHS_IN_QUARTER
         interval = 'month'
       (@_advanceMiddle(tick, interval, count) for tick in ticks)
-  .property 'xDomain', 'maxNumberOfLabels', 'centerAxisLabels', 'selectedInterval'
+  .property 'xDomain', 'centerAxisLabels', 'xAxisTimeInterval'
 
   _advanceMiddle: (time, interval, count) ->
     new Date (time = time.getTime()/2 + d3.time[interval].offset(time, count)/2)
 
-  # the years which should be labelled
-  labelledYears: (start, stop) ->
-    years = d3.time.years(start, stop)
+  # The amount of time between a start and a stop in the specified units
+  # Note that the d3 time library was not used to calculate all of these times
+  # in order to improve runtime. This comes at the expense of accuracy, but for
+  # the applications of timeBetween, it is not too important
+  numTimeBetween : (timeInterval, start, stop) ->
+    switch timeInterval
+      when 'seconds' then (stop - start)/1000
+      when 'hours' then (stop - start)/1000
+      when 'days' then (stop - start)/86400000
+      when 'weeks' then  d3.time.weeks(start, stop).length
+      when 'months' then d3.time.months(start, stop).length
+      when 'quarters' then d3.time.months(start, stop).length/MONTHS_IN_QUARTER
+      when 'years' then d3.time.years(start, stop).length
 
-    # if we have too many labelled years then we reduce to the maximum
-    # number of years labelled such that they are uniformly distributed
-    # in the range
-    if years.length > @get('maxNumberOfLabels')
-      skipVal = Math.ceil(years.length / @get('maxNumberOfLabels'))
-      d3.time.years(start, stop, skipVal)
-    else
-      years
+  # The labeller type used to create the labels for each domain type
+  # Note that quarters uses a month labeller to create the labels
+  domainTypeToLabellerType =
+    'S': 'seconds'
+    'H': 'hours'
+    'D': 'days'
+    'W': 'weeks'
+    'M': 'months'
+    'Q': 'months'
+    'Y': 'years'
 
-  # the quarters which should be labelled
-  labelledQuarters: (start, stop) ->
-    quarters = d3.time.months(start, stop, 3)
+  # The lengthened representation for each domain type. This is different from
+  # domainTypeToLabellerType
+  domainTypeToLongDomainType =
+    'S': 'seconds'
+    'H': 'hours'
+    'D': 'days'
+    'W': 'weeks'
+    'M': 'months'
+    'Q': 'quarters'
+    'Y': 'years'
 
-    # if we have too many quarters then we only display quarter labels
-    # on years
-    if quarters.length > @get('maxNumberOfLabels')
-      @labelledYears(start, stop)
-    else
-      quarters
+  domainTypeToLongDomainTypeSingular: (timeInterval) ->
+    domainType = domainTypeToLongDomainType[timeInterval]
+    domainType.substring(0, domainType.length - 1)
```

Comment:
luckily the plurals of all time things are just `s` :p that said I'm fine with this